Another round of minor specification review fixes (#1126)
riscv/riscv-cheri: src/cheri/riscv-unpriv-integration.adoc

. `Y` denotes a CHERI base.
. `A` denotes the capability encoding format.

> NOTE: This naming scheme is based on the RISC-V https://lists.riscv.org/g/tech-unprivileged/topic/longer_base_name_proposal/116854896["longer base name" proposal], which allows base ISA names longer than two characters.

Reviewer: do we need this?

{cheri_base_ext_name} specifies the minimum required fields that the capability encoding format must support, and their semantics.
This version of {cheri_base_ext_name} defines only little-endian capability memory layouts.

> The memory representation of capabilities is always `2*XLEN` (`YLEN`) bits wide and holds the address in the first `XLEN` bits of memory followed by the metadata.

Reviewer: ```suggestion
The memory representation of capabilities is always `2*XLEN` (`YLEN`) bits wide and holds the address in the lower `XLEN` bits of memory and the metadata in the upper `XLEN` bits.
```

* Any store that wrote the capability to memory was correctly authorized.
* Any load that read the capability from memory was correctly authorized.

> When an operation fails a check, the operation raises an exception or sets the resulting {ctag} to zero, as defined by each instruction.

Reviewer: I quite liked "either due to software error or malicious intent"

{ctag_cap}s are tracked through the memory subsystem: every aligned YLEN-bit wide region has a non-addressable one-bit {ctag}, which the hardware manages atomically with the data.
The implementation must set the {ctag} to zero if any byte in the YLEN/8 aligned memory region is ever written using an operation other than a store of a capability operand.

> For example, the <<STORE_CAP>> instruction sets the {ctag} to one if all the instruction's prerequisites are met, but <<ldst,SW>> will always set the {ctag} to zero.

Reviewer: ```suggestion
For example, the <<STORE_CAP>> instruction sets the {ctag} to one if all the instruction's prerequisites are met, including `{cs2}` 's {ctag} being set to one, but <<ldst,SW>> will always set the stored {ctag} to zero.
```

 This scheme could represent all out-of-bounds pointers with a very high hardware cost.

Since the upper bits of the address are used to calculate the top and base bounds, deriving a new capability with a different address could change the resulting bounds.
All instructions that derive a capability with a new address must check that the new address does not change the bounds.

> If the bounds _do_ change then the {ctag} of the derived capability is set to zero, so that it is invalid for use.

Reviewer: ```suggestion
If the bounds _do_ change then the {ctag} of the derived capability is set to zero, so that the capability becomes invalid.
```
